pvlib/modelchain.py

```diff
@@ -379,10 +379,19 @@ def dc_model(self, model):
                                          str(missing_params))
                 if model == 'sapm':
                     self._dc_model = self.sapm
-                elif model == 'singlediode':
-                    self._dc_model = self.singlediode
+                elif model == 'desoto':
+                    self._dc_model = self.desoto
+                elif model == 'pvsyst':
+                    self._dc_model = self.pvsyst
                 elif model == 'pvwatts':
                     self._dc_model = self.pvwatts_dc
+                elif model == 'singlediode':
+                    warnings.warn('DC model keyword singlediode used for '
+                                  'ModelChain object. singlediode is '
+                                  'ambiguous, use desoto instead. singlediode '
+                                  'keyword will be removed in v0.7.0 and '
+                                  'later', DeprecationWarning)
```

Comment:
This should be a ``pvlibDeprecationWarning``.

Can you also look into adding a test that will fail on pvlib versions >= 0.7? See [test_irradiance.py::test_deprecated_07](https://github.com/pvlib/pvlib-python/blob/master/pvlib/test/test_irradiance.py#L71-L80) for an example.

Reply:
Near the top of both ``modelchain.py`` and ``test_modelchain.py``:

``from pvlib._deprecation import pvlibDeprecationWarning``

On line 22 of ``test_modelchain.py``: 

``from conftest import fail_on_pvlib_version, requires_scipy``

Reply:
Done